frontend/packages/metal3-plugin/src/components/host-status.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import { Button } from 'patternfly-react';
+
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { StatusIconAndText } from '@console/internal/components/utils/status-icon';
+import { getHostStatus } from '../utils/host-status';
+
+import {
+  HOST_STATUS_DISCOVERED,
+  HOST_PROGRESS_STATES,
+  HOST_ERROR_STATES,
+  HOST_SUCCESS_STATES,
+} from '../constants';
+
+// TODO(jtomasek): Update this with onClick handler once add discovered host functionality
+// is available
+export const AddDiscoveredHostButton: React.FC<{ host: K8sResourceKind }> = (
+  { host }, // eslint-disable-line @typescript-eslint/no-unused-vars
+) => (
+  <Button bsStyle="link">
+    <StatusIconAndText status="Add host" iconName="add-circle-o" />
+  </Button>
+);
+
+type BaremetalHostStatusProps = {
+  host: K8sResourceKind;
+  machine?: K8sResourceKind;
+  node?: K8sResourceKind;
+};
+
+const BaremetalHostStatus = ({ host }: BaremetalHostStatusProps) => {
+  const hostStatus = getHostStatus(host);
+  const { status, title } = hostStatus;
+
+  switch (true) {
```

Comment:
Clever, but is this really better than...?

```ts
if (status === HOST_STATUS_DISCOVERED) {
  return <AddDiscoveredHostButton host={host} />;
}
if (HOST_PROGRESS_STATES.includes(status)) {
  return <StatusIconAndText status={title} iconName="refresh" />;
}
// ...
```

Reply:
No strong opinion, but I find the switch statement a but more readable